src/theme-provider.js

```diff
@@ -29,6 +30,7 @@ class ThemeProvider extends Component {
 
   setOuterTheme = theme => {
     this.outerTheme = theme
+    this.broadcast.setState(this.getTheme())
```

Comment:
Uhm...it would still te-render because if the emitter broadcasted an update, the parent provider would setState, and the inner would received new props, triggering a children te-render. I know it's a bit confusing. Probably your latest commit was not necessary. BTW I think I wrote a test covering that specific case